src/test/unit_tests/different_clients/go_aws_sdkv2_client.go

```diff
@@ -0,0 +1,345 @@
+// go run ./src/test/unit_tests/different_clients/go_aws_sdkv2_client.go -bucket second.bucket -key test-key -mpu test-mpu-key -endpoint "https://localhost:12443" -disable-deletion
+func main() {
+	printTestIntroduction()
+	flag.Parse()
+	checkRequiredFlags()
+	client := configureS3Client()
+	runTests(client)
+	deleteObjectsAndBucket(client)
+	printTestSummary()
+}
+
+// readEnv will read the environment variable and return its value
+func readEnv(envVar string) string {
+	value, exists := os.LookupEnv(envVar)
+	if !exists {
+		fmt.Printf("Env %s is not set\n", envVar)
+	}
+	return value
+}
+
+// checkRequiredFlag will check if the flag is empty and print the default flags
+func checkRequiredFlag(flagName string) {
+	if len(flagName) == 0 {
+		flag.PrintDefaults()
+		log.Fatalf("invalid parameters, %s name required", flagName)
+	}
+}
+
+// checkRequiredFlags will check if each of the required flags is empty
+func checkRequiredFlags() {
+	checkRequiredFlag(bucket)
+	checkRequiredFlag(key)
+	checkRequiredFlag(mpu)
+}
+
+// printTestIntroduction will print the introduction of the test
+func printTestIntroduction() {
+	fmt.Println("Running a couple of tests using AWS SDK Go V2...")
+	fmt.Println("--------------------------------------------------")
+}
+
+// printTestSummary will print the summary of the test (passing, failing and skipped tests)
+func printTestSummary() {
+	totalTests := len(passingTests) + len(failingTests)
+	fmt.Println("--------------------------------------------------")
+	fmt.Println("\nTotal Tests:", totalTests,
+		"\nPassing Tests: ", len(passingTests), passingTests,
+		"\nFailing Tests: ", len(failingTests), failingTests,
+		"\nSkipped Tests: ", len(skippedTests), skippedTests,
+	)
+}
+
+// runTests will run the tests
+// according to the following list: Create Bucket, Put Object, Put Multipart Objects
+// the deletion operation (Delete Object, Delete Multipart Upload Object and Delete Bucket) were moved to a separate function
+func runTests(client *s3.Client) {
+	createBucket(client, bucket)
+	putObject(client, bucket, key)
+	uploadId := createMultipartUpload(client, bucket, mpu)
+	if uploadId != nil {
+		completedParts := uploadPart(client, bucket, mpu, uploadId)
+		if completedParts != nil {
+			completeMultiPartUpload(client, bucket, mpu, uploadId, completedParts)
+		} else {
+			skippedTests = append(skippedTests, "completeMultiPartUpload")
+		}
+	} else {
+		skippedTests = append(skippedTests, "uploadPart")
+		skippedTests = append(skippedTests, "completeMultiPartUpload")
+	}
+}
+
+// deleteObjectsAndBucket will delete the 2 created objects (one from Put Objects and the other from Multipart Upload Objects) and the bucket
+func deleteObjectsAndBucket(client *s3.Client) {
+	if disableDeletion {
+		fmt.Println("--------------------------------------------------")
+		fmt.Println("disable-deletion flag was set, will not operate delete commands")
+	}
```

Comment:
I think we can return here, if deletion is disabled to avoid later checks.

Reply:
I added an else clause, thanks.